### PR TITLE
Update variables in HRVO to use snake case

### LINKED ISSUE #2717: Update variables in HRVO to use snake case
### Description of the task

HRVO is an external path planning implementation we use for navigating around moving obstacles and we'd like to maintain our own copy of it. Currently, the external implementation doesn't follow our code guidelines and we'd like to update it so that it does.

One major problem is that the external implementation uses camelCase for variable names while we use snake_case. This ticket would involve converting these variables into snake case. For example (`distSqLeft` in `kd_tree.cpp` would become `dist_sq_left`, `neightborDist` would become `neighbour_dist` in `simulator.cpp`)

### Acceptance criteria

- [ ] Navigate to `Software/src/extlibs/hrvo` and browse the header and source files
- [ ] Identify offending variables that use camelCase and update them to use snake_case
- [ ] Fix any build issues that come up when running `bazel build //software/...` in the Software/src directory

### PR DESCRIPTION
**Pull request by germchow**
### Description
fixed the variable names inside all files in `Software/src/extlibs/hrvo` by making them snake case.

### Testing Done
All 8 tests in `//extlibs/hrvo:simulated_hrvo_test` pass

### Resolved Issues
Resolves #2717 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/extlibs/hrvo/hrvo_agent.h [hunk 1]
```diff
@@ -144,11 +144,11 @@ class HRVOAgent : public Agent
...
 
-        // The index of the first VelocityObstacle in velocityObstacles_ used to compute
+        // The index of the first VelocityObstacle in velocity_obstacles_ used to compute
         // this candidate.
         int velocityObstacle1_;
 
-        // The index of the second VelocityObstacle in velocityObstacles_ used to compute
+        // The index of the second VelocityObstacle in velocity_obstacles_ used to compute
         // this candidate.
         int velocityObstacle2_;
```
Comment (nimazareian) [inline src/extlibs/hrvo/hrvo_agent.h]: These two variables should also be snake case

File: src/extlibs/hrvo/hrvo_agent.cpp [hunk 2]
```diff
@@ -67,7 +68,7 @@ void HRVOAgent::computeNeighbors()
     Vector current_dest = path.getCurrentPathPoint().value().getPosition();
 
     float new_neighbor_dist =
-        std::min(static_cast<double>(neighborDist_),
+        std::min(static_cast<double>(neighbor_dist_),
```
Comment (nimazareian) [inline src/extlibs/hrvo/hrvo_agent.cpp:71]: GitHub doesn't allow me to add a comment on line 88, but the variable `openingAngle` should also be updated.

File: src/extlibs/hrvo/kd_tree.cpp [hunk 3]
```diff
@@ -154,72 +154,72 @@ void KdTree::queryRecursive(HRVOAgent *agent, float &rangeSq, std::size_t node)
...
-            distSqRight +=
+            dist_sq_right +=
                 std::pow(agent->position_.y() - nodes_[nodes_[node].right_].maxY_, 2.f);
         }
 
-        if (distSqLeft < distSqRight)
+        if (dist_sq_left < dist_sq_right)
         {
-            if (distSqLeft < rangeSq)
+            if (dist_sq_left < rangeSq)
```
Comment (nimazareian) [inline src/extlibs/hrvo/kd_tree.cpp]: `rangeSq` should also be updated

**Review by nimazareian — CHANGES_REQUESTED** [review]
Thanks for going through the task quickly and creating a PR! 
I searched through the files using this regex `[a-z]+[A-Z][a-z]+` (by all means could be improved) and noticed a couple of variables which should also be updated.

File: src/extlibs/hrvo/hrvo_agent.h (hunk 1 above)
Comment (germchow) [reply src/extlibs/hrvo/hrvo_agent.h]: I see thanks (I didn't know I could use regex to search through files so I just did it manually).

I found that `[a-zA-Z][ ][a-z]+[A-Z][a-z]+` worked good for finding camelcase variables, I'll pass it on to the other new members working on similar tasks

**Comment by nimazareian** [comment]
The one test failing in CI seems to be undeterministic, so I have removed it in #2735.

**Review by nimazareian — APPROVED** [review]
LGTM!

**Review by JoshuaWu7 — APPROVED** [review]

**Pull request merged**